frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  Button,
+  useToast,
+} from "@chakra-ui/react";
+import {
+  Camper,
+  EditCamperInfoFields,
+  EditModalSetterFunctions,
+} from "../../../../types/CamperTypes";
+import { FormQuestion } from "../../../../types/CampsTypes";
+import CamperApiClient from "../../../../APIClients/CamperAPIClient";
+import EditCamperModalForm from "./EditCamperModalForm";
+
+type EditCamperModalProps = {
+  camper: Camper;
+  editCamperModalIsOpen: boolean;
+  formQuestions: FormQuestion[];
+  editCamperOnClose: () => void;
+};
+
+const EditCamperModal = ({
+  camper,
+  formQuestions,
+  editCamperModalIsOpen,
+  editCamperOnClose,
+}: EditCamperModalProps): JSX.Element => {
+  const toast = useToast();
+
+  // State for the edit form fields
+  const [firstName, setFirstName] = useState<string>(camper.firstName);
+  const [lastName, setLastName] = useState<string>(camper.lastName);
+  const [age, setAge] = useState<number>(camper.age);
+  const [allergies, setAllergies] = useState<string | undefined>(
+    camper.allergies,
+  );
+  const [specialNeeds, setSpecialNeeds] = useState<string | undefined>(
+    camper.specialNeeds,
+  );
+  const [formResponses, setFormResponses] = useState<
+    Map<string, string> | undefined
+  >(camper.formResponses);
+
+  const setStateFuncs: EditModalSetterFunctions = {
+    setFirstName,
+    setLastName,
+    setAge,
+    setAllergies,
+    setSpecialNeeds,
+    setFormResponses,
+  };
+
+  const stateVariables: EditCamperInfoFields = {
+    hasPaid: camper.hasPaid,
+    firstName,
+    lastName,
+    age,
+    allergies,
+    specialNeeds,
+    formResponses,
+  };
+
+  const onSaveClicked = async () => {
+    const editCamperFields: EditCamperInfoFields = {
+      firstName,
+      lastName,
+      age,
+      allergies,
+      specialNeeds,
+      formResponses,
+      hasPaid: camper.hasPaid,
+    };
+    const isEditCamperSuccess = await CamperApiClient.updateCampersById(
+      [camper.id],
+      editCamperFields,
+    );
+
+    if (isEditCamperSuccess) {
+      // TODO: introduce global state so we don't have to reload
+      toast({
+        description: `${camper.firstName} ${camper.lastName}'s information has been updated`,
+        status: "success",
+        variant: "subtle",
+        duration: 3000,
+      });
+
+      setTimeout(() => {
+        window.location.reload();
+      }, 3000);
```

Comment:
Can we move down from 3 secs? Even as the brute force approach - maybe we can't guarantee that we'll cover everyone's network speed, but I suspect this is going to feel like an error for most reasonable users

Reply:
changed it to 2 seconds